chain-core/src/property.rs

```diff
@@ -307,11 +306,29 @@ pub trait Serialize {
 
 /// Define that an object can be read from a `Read` object.
 pub trait Deserialize: Sized {
-    type Error: std::error::Error + From<std::io::Error>;
+    type Error: std::error::Error + From<std::io::Error> + Send + Sync + 'static;
 
     fn deserialize<R: std::io::BufRead>(reader: R) -> Result<Self, Self::Error>;
 }
 
+pub trait FromStr: Sized {
+    type Error: std::error::Error + Send + Sync + 'static;
+
+    fn from_str(s: &str) -> Result<Self, Self::Error>;
+}
+
+impl<T> FromStr for T
+where
+    T: std::str::FromStr,
+    <T as std::str::FromStr>::Err: std::error::Error + Send + Sync + 'static,
+{
+    type Error = <T as std::str::FromStr>::Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Error> {
+        std::str::FromStr::from_str(s)
+    }
+}
```

Comment:
argh... this is very hacky... :/

Reply:
https://github.com/rust-lang/rust/issues/32722 does not allow just adding a subtrait with extra bounds on `FromStr::Err`.